torch/_functorch/aot_autograd.py

```diff
@@ -3850,7 +3935,7 @@ def forward(ctx, *deduped_flat_tensor_args):
                 fw_outs[num_forward_returns:num_forward],
                 return_new_outs=False
             )
-            return tuple(raw_returns)
+            return tuple(raw_returns) + tuple(marked_dirty_inps)
```

Comment:
@albanD How come inputs which are `mark_dirty` have to be explicitly returned as outputs to the autograd function? Seems a little dorky to me

Reply:
It is necessary in the case where the input is a view so that we can do the proper graph rewrite.
But I agree that in this case, it does not help. We can skip this check when there are multiple outputs (since if the input was a view, it would have failed before). Do you want me to send a PR?

Reply:
In theory, AOTAutograd would still need to handle this even if we skipped the check when there are multiple outputs, right? Since if your compiled function happens to mutate a single input and return nothing, then we would still be forced to do the mark_dirty logic.

So I guess skipping the check in most cases could be nice in that it would help us avoid some runtime overhead (not having to manipulate python lists in the hotpath to filter the mark_dirty tensors in most cases). But we still need some conditional logic in AOTAutograd to "sometimes" return mark-dirty inputs. e.g:

```
@torch.compile
def f(x)
    # x will be marked dirty, and autograd.Function will still require us to return it as an output here
    x.mul_(2)
    return
```

Reply:
Yes that is correct.